### DISCUSSION TITLE
Close binary system

### DISCUSSION THREAD

**Discussion by ziluyang9421** [phoebe-project/phoebe2#886] [Questions]
Hello everyone!

I need some help. I have just started studying binary stars, specifically close binary stars composed of a white dwarf and a main sequence star. I want to use Phoebe to obtain the orbital parameters of the close binary. However, at the beginning, I confused close binaries with contact binaries, so I wrote my code following the example of contact binaries in the tutorial. When I tried to change the eccentricity (ecc), an error occurred saying that the orbit of a contact binary must be circular and ecc needs to be 0. But there is a significant ellipsoidal variation in the light curve, and there may even be spots.

Could you please suggest which example is more appropriate for close binaries with ellipsoidal variations? What parameters should I pay special attention to? If I want to add spots (spots changing over time), how should I do it?

Thank you very much!

**Comment by kecnry** [comment]
Welcome!  Contact binaries can still show (significant) ellipsoidal variations, but if you are sure that your system is definitely eccentric, then it is unlikely to be in contact.  If that's the case, then you want two detached stars that are _close_ enough to each other to cause significant tidal distortions but without actually making contact by using a detached binary system `phoebe.default_binary()`, leaving `distortion_method = 'roche'` (the default value), and probably leaving `irradition_method` enabled.  You'll need to ensure that the stars never overflow at closest approach/periastron (phoebe will check this for you and complain if they do, but you can also look at the `requiv` and `requiv_max` parameters - `requiv` can never extend beyond `requiv_max`).  There is a chance that some of your surface elements will extend beyond the range of the default atmosphere table (ck2004), especially for the white dwarf.  If/when that happens, you can change the value of `atm` and may also need to change limb-darkening (see `ld_mode` and `ld_mode_bol` parameters) from 'lookup' to 'manual' and then provide coefficients manually (see `ld_coeffs` and `ld_coeffs_bol` parameter).

For spots, you can see [this tutorial](http://phoebe-project.org/docs/latest/tutorials/spots) - but note that phoebe does not currently support time-dependent spots.  You could always change the values and re-compute the synthetic model in different time chunks, but that can get clunky (see #877 for a discussion on a similar case).

Hopefully this at least helps you get started!  Let us know if you run into any problems or questions.

-Kyle

**Reply by ziluyang9421** [reply]
Thank you for your guidance!!! I know what to do next. If I have any questions, I will come back for help.
